src/middlewares/auth.middleware.js

```diff
@@ -1,6 +1,32 @@
 const User = require('../models/user.models')
 const {response_400, response_500} = require('../utils/responseCodes.utils')
+const sendError=require('../utils/responseCodes.utils.js');
+const jwt = require('jsonwebtoken');
+const secretKey = "777";
+const generateToken=User.methods.generatePasswordReset();
+function authorizationMiddleware(req, res, next) {
+    const token = req.cookies.token;
 
+    if (!token) {
+        return sendErrorResponse(res, 400 , 'Forbidden', 'No token provided');
+    }
+
+    try {
+        const decoded = jwt.verify(token, secretKey);
+        req.userId = decoded.userId;
```

Comment:
decoded itself is userId (mongo id for user). find the user by querying the db and add it to req.user